src/ApiService/ApiService/onefuzzlib/VmssOperations.cs

```diff
@@ -411,19 +410,21 @@ private Task<string> GetInstanceIdForVmId(Guid scaleset, Guid vmId)
         }
     }
 
-    public async Task<List<string>?> ListVmss(Guid name, Func<VirtualMachineScaleSetVmResource, bool>? filter) {
+    public async Async.Task<List<string>?> ListVmss(Guid name, Func<VirtualMachineScaleSetVmResource, bool>? filter) {
         try {
-            var vmss = await _creds.GetResourceGroupResource().GetVirtualMachineScaleSetAsync(name.ToString());
-            return vmss.Value.GetVirtualMachineScaleSetVms()
+            // have to assign to a typed variable so that the Where call can be disambiguated
+            IAsyncEnumerable<VirtualMachineScaleSetVmResource> vms =
```

Comment:
how about we return the asyncenumerable list and let the caller do the filtering. 
We won't need to consume the full list here.

Reply:
done! There's only one caller so it was an easy change.